components/timeline/src/Timeline.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC, ReactNode, createElement as h } from 'react';
+import { StandardProps, classBuilder } from '@not-govuk/component-helpers';
+
+import '../assets/Timeline.scss';
+
+type TimelineItem = {
+  /** date is used to set the datetime attribute of the <time> element so must
+  be in the ISO format for a date or time. */
+  action: string
+  by?: string
+  date: Date
+  description: string
```

Comment:
Would be nice to have comments for all of them but I'm not too fussed as I'm not sure they would show up in the docs at the moment. (As it would only show `TimelineProps`.)